components/org.wso2.carbon.identity.api.server.action.management/org.wso2.carbon.identity.api.server.action.management.v1/src/main/java/org/wso2/carbon/identity/api/server/action/management/v1/core/ServerActionManagementService.java

```diff
@@ -374,14 +376,16 @@ private Authentication.Type getAuthTypeFromPath(String authType) {
                         ERROR_INVALID_ACTION_ENDPOINT_AUTH_TYPE));
     }
 
-    /**
-     * Handle not implemented action types.
-     *
-     * @param actionType Action type.
-     */
-    private void handleNotImplementedActionTypes(String actionType) {
+    private void validateActionType(String actionType) {
+
+        Action.ActionTypes actionTypeEnum =
+                Arrays.stream(Action.ActionTypes.filterByCategory(Action.ActionTypes.Category.PRE_POST))
+                        .filter(actionTypeObj -> actionTypeObj.getPathParam().equals(actionType))
+                        .findFirst()
+                        .orElseThrow(() -> ActionMgtEndpointUtil.handleException(Response.Status.BAD_REQUEST,
+                                ERROR_INVALID_ACTION_TYPE));
```

Comment:
Do we need to do this validation in the api layer as this is already validated in the service layer ?
https://github.com/wso2/carbon-identity-framework/blob/9c99ba529c944312102845d7cdc9c50475b3b97d/components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/ActionManagementServiceImpl.java#L275C1-L281C6

Reply:
This validates against PRE_POST action type only. Service validation is against all action types